src/main/java/org/broadinstitute/hellbender/utils/SimpleInterval.java

```diff
@@ -14,203 +15,238 @@
-    public static final char START_END_SEPARATOR = '-';
-    public static final String END_OF_CONTIG = "+"; //note: needs to be a String because it's used in an endsWith call.
-
-    private final int start;
-    private final int end;
-    private final String contig;
-
-    /**
-     * Create a new immutable 1-based interval of the form [start, end]
-     * @param contig the name of the contig, must not be null
-     * @param start  1-based inclusive start position
-     * @param end  1-based inclusive end position
-     */
-    public SimpleInterval(final String contig, final int start, final int end){
-        validatePositions(contig, start, end);
-        this.contig = contig;
-        this.start = start;
-        this.end = end;
-    }
-
-    /**
-     * Create a new SimpleInterval from a {@link Locatable}
-     * @param locatable any Locatable
-     * @throws IllegalArgumentException if locatable violates any of the SimpleInterval constraints or is null
-     */
-    public SimpleInterval(final Locatable locatable){
-        this(Utils.nonNull(locatable).getContig(),
-                locatable.getStart(), locatable.getEnd());
-    }
-
-    /**
-     * Test that these are valid values for constructing a SimpleInterval
-     * @throws IllegalArgumentException if it is invalid
-     */
-    private static void validatePositions(final String contig, final int start, final int end) {
-        if ( contig == null ) {
-            throw new IllegalArgumentException("contig cannot be null");
-        }
-        if ( start <= 0 ) {
-            throw new IllegalArgumentException("SimpleInterval is 1 based, so start must be >= 1, start:%d" + start);
-        }
-        if ( end < start ) {
-            throw new IllegalArgumentException(String.format("end must be >= start. start:%d end:%d", start, end));
-        }
-    }
-
-    /**
-     * Makes an interval by parsing the string.
-     *
-     * @warning this method does not fill in the true contig end values
-     * for intervals that reach to the end of their contig,
-     * uses {@link Integer#MAX_VALUE} instead.
-     *
-     * Semantics of start and end are defined in {@link Locatable}.
-     * The format is one of:
-     *
-     * contig           (Represents the whole contig, from position 1 to the {@link Integer#MAX_VALUE})
-     *
-     * contig:start     (Represents the 1-element range start-start on the given contig)
-     *
-     * contig:start-end (Represents the range start-end on the given contig)
-     *
-     * contig:start+    (Represents the prefix of the contig starting at the given start position and ending at {@link Integer#MAX_VALUE})
-     *
-     * examples (note that _all_ commas in numbers are simply ignored, for human convenience):
-     *
-     * 'chr2', 'chr2:1000000' or 'chr2:1,000,000-2,000,000' or 'chr2:1000000+'
-     */
-    public SimpleInterval(final String str){
+     private static final long serialVersionUID = 1L;
```

Comment:
There's some sort of spacing change here.  If it's possible to revert that it would be conveinient.